lib/features/build_server/compiler_interface.dart

```diff
@@ -190,13 +199,34 @@ class WorkerInterface extends Interface with CompilerStatusDelegate {
       payload: null,
     ));
   }
+
+  @override
+  void didFailCompilation() {
+    _updateState(CompileStatusMessage(
+      status: CompileStatus.failed,
+      payload: null,
+    ));
+  }
+
+  @override
+  void didFailLinking() {
+    _updateState(CompileStatusMessage(
```

Comment:
I'm not entirely warm with the idea of not separating compiler and linker messages. I fear that this will bring me some headache when I will add some features to the compiler or we change up the protocol. But what do u think? Give up some DRY for better maintainability?

Reply:
Yes, I see your point. But why do we send `CompileStatusMessage` then? Wouldn't it make sense to send something called `LinkStatusMessage`? Or do you want to unite them as they are very similar? Maybe call it `BuildStatusMessage` if that's what you intend :)